test/increment-compression.test.ts

```diff
@@ -0,0 +1,30 @@
+import {v4} from 'uuid';
+
+import {SetupIntegrationTest} from './integration-setup';
+
+const {client} = SetupIntegrationTest();
+
+describe('increment, with compression client', () => {
+  it('should return error saying compression not supported', async () => {
+    if (process.env.COMPRESSION === 'true') {
```

Comment:
2 nits here:

1. By referencing the environment variable here you're breaking down our separation of concerns a bit. There are now several places in the code that have to have this knowledge that the env var is how we are controlling the compression, and if we decided to change that in the future we would have to make sure to find all of those places in the code and update them, or else we might let bugs slip through. It's better to centralize this configuration choice to only one spot in the code. That's why I was suggesting that you extend `SetupIntegrationTest` to also return the value for `useCompression` - then we only have exactly one spot in the code where we need to know about the env var.
2. I think I'd probably change this logic around to `if compression = false { log('skipping this test because x'); return; }`. it just keeps the condition blocks shorter so that there's less stuff people need to keep in their head when looking at an individual line of code. this is kind of a stylistic preference though, so definitely not a blocker for the PR if you disagree!